src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -156,27 +156,10 @@ void QgsLayoutItemLegend::refresh()
 
 void QgsLayoutItemLegend::draw( QgsLayoutItemRenderContext &context )
 {
-  QPainter *painter = context.renderContext().painter();
-  painter->save();
-
-  // painter is scaled to dots, so scale back to layout units
-  painter->scale( context.renderContext().scaleFactor(), context.renderContext().scaleFactor() );
-
-  painter->setPen( QPen( QColor( 0, 0, 0 ) ) );
-
-  if ( !mSizeToContents )
-  {
-    // set a clip region to crop out parts of legend which don't fit
-    QRectF thisPaintRect = QRectF( 0, 0, rect().width(), rect().height() );
-    painter->setClipRect( thisPaintRect );
-  }
-
   QgsLegendRenderer legendRenderer( mLegendModel.get(), mSettings );
   legendRenderer.setLegendSize( mSizeToContents ? QSize() : rect().size() );
 
-  legendRenderer.drawLegend( painter );
-
-  painter->restore();
+  legendRenderer.drawLegend( context.renderContext() );
```

Comment:
Move above the restore call - the painter restore needs to be brought back, and happen after the drawLegend call

Reply:
Can I combine the two fixes and simply add

`context.renderContext().painter()->restore()`

right below the drawLegend call?

Reply:
Yes -- it's effectively the same, because painter is just a pointer to `context.renderContext().painter()`. So either call applies to the same object.